.github/workflows/riot-ryot.yml

```diff
@@ -0,0 +1,94 @@
+# Run 'compile_and_test_for_board.py' on connected boards.
+#
+# This workflow will run on a RYOT machine and lunch all tests on all boards
+# currently connected to that machine. Builds can be parallelized if multiple
+# github self-hosted runners are configured.
+#
+# Running self-hosted runners on public repositories is not advised, but here
+# we do not run them on PRs, only on master and release candidates.
+#
+# Documentation:
+#
+# * Setup a RYOT machine:
+#   https://github.com/fjmolinas/riot-ryot/blob/master/setup.md
+# * Setup Runner:
+#   https://github.com/fjmolinas/riot-ryot/blob/master/setup.md#self-hosted-github-runners
+
+name: riot-ryot
+
+on:
+  # Schedule weekly runs Saturday at 00:00 on master
+  schedule:
+    - cron: '00 0 * * 6'
+  push:
+    # TODO: remove only here to demonstrate build once
+    branches:
+      - master
+    # Run on all new release candidates
+    tags:
+      - '*-RC*'
+
+jobs:
+  # Get all currently connected boards
+  connected_boards:
+    name: Get Connected Boards
+    runs-on: self-hosted
+    env:
+      RIOT_MAKEFILES_GLOBAL_PRE: /builds/conf/makefiles.pre
+    outputs:
+      boards: ${{ steps.list-boards.outputs.boards }}
+    steps:
+      - id: list-boards
+        run: echo "::set-output name=boards::$(make -C /builds/boards/ list-boards-json --no-print-directory)"
+
+  # Runs all tests on connected boards
+  compile_and_test_for_board:
+    name: ${{ matrix.board }}
+    runs-on: self-hosted
+    needs: connected_boards
+    continue-on-error: true
```

Comment:
The problem with this option is that even if there are failed jobs in the matrix, the global status remains "success". You can set `fail-fast: false` under the `strategy` group as a work-around but I'm not that in this case the next job (`results_summary`) would run.

Reply:
Hmm I guess I can write the output to a file and read that at the end to send the final job failure.

Reply:
Or simply read `failures_summary`

Reply:
or you can drop the `results_summary` job which I find a bit useless since artifacts could be pushed for each board (see how it's done in #14587).

Reply:
Well I don't find it useless.

Reply:
Having a global picture is usefull.

Reply:
The global picture is already provided by the github interface: it shows on which board something was broken. Then, using a per-board artifact, one can concentrate on the board which reported issues. (I hope one day compile_an_test_board will succeed without errors).
It's also super easy to know what was broken on a given board by opening the archive and check the summary. This can be done in 3 clicks.

Here, adding an extra step for gathering all board results:
- adds extra complexity to the workflow: for retrieving the compile_and_test_for_board job status and pass it to the next job, for exposing the artifact of each board in the matrix to the result_summary job.
- requires an extra tool to gather the results in a single artifact

Let's keep it simple and focus on what is really necessary: run compile_and_test_for_board on a list of boards. We can still use the github interface to share the results.

Reply:
Ok, will address when I have time then.

Reply:
addressed